velox/type/tests/FilterTest.cpp

```diff
@@ -898,3 +898,27 @@ TEST(FilterTest, mergeWithBigintMultiRange) {
     }
   }
 }
+
+TEST(FilterTest, mergeMultiRange) {
+  std::vector<std::unique_ptr<Filter>> filters;
+  addUntypedFilters(filters);
+
+  filters.push_back(
+      orFilter(lessThanOrEqualFloat(1.2), betweenFloat(-1.2, 3.4)));
```

Comment:
The filters in MergeRange are expected to not overlap. Let's change this to <= 1.2 OR between 1.2 and 3.4.